containerd-shim/main.go

```diff
@@ -106,7 +106,7 @@ func start(log *os.File) error {
 		case s := <-signals:
 			switch s {
 			case syscall.SIGCHLD:
-				exits, _ := osutils.Reap()
+				exits, _ := osutils.Reap(true)
```

Comment:
So it doesn't use `WNOHANG`. We received a `SIGCHLD` so there's no risk in blocking until it is ready to be reaped.

Initially I remove the possibility of this routine being non-blocking, but since it could technically be used elsewhere, I added this parameter.